contracts/Cartographer.sol

```diff
@@ -941,14 +942,42 @@ contract Cartographer is Ownable, Initializable, ReentrancyGuard {
     }
 
 
+    /// @dev Elevate SUMMIT from the Elevation farms to the Expedition without paying any withdrawal tax
+    /// @param _elevation Elevation to elevate from
+    /// @param _amount Amount of SUMMIT to elevate
+    function elevateAndLockStakedSummit(uint8 _elevation, uint256 _amount)
+        public
+        nonReentrant poolExists(address(summit), _elevation)
+    {
+        require(_amount > 0, "Elevate non zero amount");
+
+        // Withdraw {_amount} of {_token} from {_elevation} pool
+        uint256 elevatedAmount = subCartographer(_elevation)
+            .withdraw(
+                address(summit),
+                _amount,
+                msg.sender,
+                true
+            );
+
+        // Lock withdrawn SUMMIT for EVEREST
+        expeditionV2.lockElevatableSummit(
+            elevatedAmount,
+            msg.sender
+        );
+
+        emit ElevateAndLockStakedSummit(msg.sender, _elevation, _amount);
+    }
+
+
     /// @dev Validation step of Elevate into separate function
     /// @param _token Token to elevate
     /// @param _sourceElevation Elevation to withdraw from
     /// @param _targetElevation Elevation to deposit into
     /// @param _amount Amount to elevate
     function validateElevate(address _token, uint8 _sourceElevation, uint8 _targetElevation, uint256 _amount)
-        internal
-        nonReentrant poolExists(_token, _sourceElevation) poolExists(_token, _targetElevation)
+        internal view
```

Comment:
there is no "return" value

Reply:
Yes it doesn't need a return, it is only extracting the require statements to a separate function to make sure the stack isn't too deep.

If anything fails the tx will be reverted so it doesn't need any return statement here

Reply:
it is because it's "view" function

Reply:
well, is there any specific reason that this function was changed to "view" function?